File: rb/features.rb
# LaunchLibrary2 SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module LaunchLibrary2Features
  def self.make_feature(name)
    case name
    when "base"
      LaunchLibrary2BaseFeature.new
    when "test"
      LaunchLibrary2TestFeature.new
    else
      LaunchLibrary2BaseFeature.new
    end
  end
end

File: py/features.py
# LaunchLibrary2 SDK feature factory

from feature.base_feature import LaunchLibrary2BaseFeature
from feature.test_feature import LaunchLibrary2TestFeature


def _make_feature(name):
    features = {
        "base": lambda: LaunchLibrary2BaseFeature(),
        "test": lambda: LaunchLibrary2TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
